extension.js

```diff
@@ -356,10 +372,29 @@ function update_height_setting(win) {
     settings.set_double('window-height', Math.min(1.0, current_height));
 }
 
+function set_window_maximize() {
+    if (!current_window)
+        return;
+
+    const should_maximize = settings.get_boolean('window-maximize');
+    if (current_window.maximized_vertically && !should_maximize)
+        current_window.unmaximize(Meta.MaximizeFlags.VERTICAL);
+    else if (!current_window.maximized_vertically && should_maximize)
+        current_window.maximize(Meta.MaximizeFlags.VERTICAL);
+}
+
+function update_window_height() {
```

Comment:
I don't like the ambiguous function name here - it's not clear what's the difference with `update_window_geometry`. Maybe `unmaximize_and_update_window_geometry()` or something like that.

Or you could create a function `unset_window_maximize()`/`disable_window_maximize()` and `.connect()` it to `changed::window-height` before `update_window_geometry()` (don't forget `disconnect()` too).

Actually I would prefer the 2nd option